neo/Network/NodeLeader.py

```diff
@@ -354,9 +354,47 @@ def AddTransaction(self, tx):
             return False
 
         if not tx.Verify(self.MemPool.values()):
-            logger.error("Veryfiying tx result... failed")
+            logger.error("Verifying tx result... failed")
             return False
 
         self.MemPool[tx.Hash.ToBytes()] = tx
 
         return True
+
+    def RemoveTransaction(self, tx):
+        """
+        Remove a transaction from the memory pool if it is found on the blockchain.
+
+        Args:
+            tx (neo.Core.TX.Transaction): instance.
+
+        Returns:
+            bool: True if successfully removed. False otherwise.
+        """
+        if BC.Default() is None:
+            return False
+
+        if not BC.Default().ContainsTransaction(tx.Hash):
+            return False
+
+        if tx.Hash.ToBytes() in self.MemPool:
+            del self.MemPool[tx.Hash.ToBytes()]
+            return True
+
+        return False
+
+    def MempoolCheckLoop(self):
+        """
+        Checks the Mempool and removes any tx found on the Blockchain
+        Implemented to resolve https://github.com/CityOfZion/neo-python/issues/703
+        """
+        txs = []
+        keys = self.MemPool.keys()
+        for i in keys:
+            tx = self.GetTransaction(i)
+            txs.append(tx)
+
+        for tx in txs:
+            res = self.RemoveTransaction(tx)
+            if res:
+                logger.info("found tx 0x%s on the blockchain ...removed" % tx.Hash)
```

Comment:
can you please change this to `debug` level. I don't want people to start screaming in #python that they keep seeing transactions being deleted from their blockchain. Oh and perhaps change the message that it's more obvious that the TX is removed from the mempool. If you'd see this line in some log file then at first sight you might actually think it's being removed from the blockchain (not that that's possible, but you get the idea)